ui/src/components/UploadStory.tsx

```diff
@@ -0,0 +1,855 @@
+          </AddCountryDiv>
+        )}
+
+      {children}
+    </AddCountryPaper>
+  );
+
+  const apiSubmitCall = (
+    fetchString: string,
+    method: string,
+    formData: FormData
+  ) => {
+    fetch(fetchString, {
+      method: method,
+      body: formData,
+      redirect: "follow",
+    })
+      .then((response) => response.text())
+      .then((result) => {
+        storySubmitDialog(result);
+      })
+      .catch((error) => console.log("Error: ", error));
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    setLoading(true);
+    setDisabled(true);
+    const formData = new FormData();
+    for (const key in formInput) {
+      if (key != "image" && key != "current_city") {
+        formData.append(key, formInput[key]);
+      }
+    }
+    formData.append("current_city", currentCity);
+
+    for (const index in tagArray) {
+      formData.append("tags", tagArray[index]);
+    }
+
+    let fetchString = "/api/story";
+    let method = "POST";
+
+    if (id) {
+      // if a new image was uploaded, add the image to the formdata
+      if (newImage == "") {
+        formData.append("image", formInput["image"]);
+      }
+      fetchString = `/api/story/${id}`;
+      method = "PUT";
+    } else {
+      formData.append("image", formInput["image"]);
+    }
+
+    formData.append("author_country", authorCountry);
+    if (authorCountry === addedCountry) {
+      const countryNameJsonBody = [
+        {
+          country_name: addedCountry,
+        },
+      ];
+
+      const jsonBody = JSON.stringify(countryNameJsonBody);
+      fetch("/api/countries", {
+        method: "POST",
+        body: jsonBody,
+      })
+        .then((response) => response.text())
+        .then(() => apiSubmitCall(fetchString, method, formData))
+        .catch((error) => console.log("Error: ", error));
+    } else {
+      apiSubmitCall(fetchString, method, formData);
+    }
+  };
+
+  const successMessage = id ? "Story Edit Success!" : "Story Upload Success!";
+  const pageTitle = id ? "Edit Story" : "Upload New Story";
+  const submitButtonText = id ? "Save Edits" : "Upload";
+
+  if (error) return <div>Error fetching tags!</div>;
+  if (errorCountries) return <div>Error fetching countries array!</div>;
+
+  return (
+    <>
+      <AppBar color="default" position="sticky">
+        <Grid container direction="row">
+          <Grid item xs={6}>
+            <UploadStoriesHeading>{pageTitle}</UploadStoriesHeading>
+          </Grid>
+          <Grid
+            container
+            item
+            xs={6}
+            direction="row"
+            alignContent="center"
+            justify="flex-end"
+          >
+            <StyledButton
+              disabled={!hasAllRequiredFields}
+              onClick={() => setIsDrawerOpen(true)}
+              color="primary"
+              variant="outlined"
+            >
+              Preview
+            </StyledButton>
+            <StyledButton
```

Comment:
The button is still disabled when I have all my fields filled out during upload. 
![image](https://user-images.githubusercontent.com/19617248/113806854-18c9c480-9731-11eb-9590-99423c66dc19.png)

Reply:
Hm, did you upload an image as well? Or if you added a country not on the list did you make sure to click the add button?

Reply:
Working fine now, I think it was b/c countries wasn't there